crates/task-impls/src/consensus.rs

```diff
@@ -1223,30 +1235,27 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                     return;
                 }
 
-                // stop polling for the received disperse after verifying it's valid
-                self.quorum_network
-                    .inject_consensus_info(ConsensusIntentEvent::CancelPollForVIDDisperse(
-                        *disperse.data.view_number,
-                    ))
-                    .await;
-
-                // Add to the storage that we have received the VID disperse for a specific view
-                if let Err(e) = self.storage.write().await.append_vid(disperse).await {
-                    error!(
-                        "Failed to store VID Disperse Proposal with error {:?}, aborting vote",
-                        e
-                    );
-                    return;
-                }
-
                 self.consensus
                     .write()
                     .await
                     .vid_shares
                     .entry(view)
                     .or_default()
-                    .insert(disperse.data.recipient_key.clone(), disperse.data.clone());
-
+                    .insert(disperse.data.recipient_key.clone(), disperse.clone());
+                if disperse.data.recipient_key != self.public_key {
+                    self.quorum_network
+                        .inject_consensus_info(ConsensusIntentEvent::PollForVIDDisperse(
```

Comment:
Why do we need to cancel the poll if it's not canceled before?

Reply:
Ah we don't I added this when debugging because I thought polling was cancelled after each VID share was received